pkg/controller/policyrecommendation/policyrecommendation_controller.go

```diff
@@ -539,3 +554,67 @@ func (r *ReconcilePolicyRecommendation) createDefaultPolicyRecommendationScope(c
 
 	return nil
 }
+
+// canCleanupLegacyNamespace determines whether the legacy "tigera-policy-recommendation" namespace
+// (which previously hosted the policy-recommendation) can be safely cleaned up.
+// It returns true only if:
+// - The new tigera-policy-recommendation deployment in "calico-system" exists and is available.
+// - Both the policyrecommendation custom resource and the TigeraStatus for 'policy-recommendation' are in the Ready state
+func (r *ReconcilePolicyRecommendation) canCleanupLegacyNamespace(ctx context.Context, logger logr.Logger) bool {
```

Comment:
I was wondering if we need this method at all. could we just always clean it up? (for multitenant cluster, there is nothing in the tigera-policy-recommendation namespace anyway.